File: js/index.test.js
import processor from ".";
import { readFile } from "node:fs";
describe("Name of the group", () => {
  test.only("should ", () => {
    expect(processor(3, "helloy")).toBe("khoorb");
  });
  // test.skip("should ", () => {
  //   let cyphered;
  //   readFile("./cyphered.txt", (err, data) => {
  //     if (err) throw err;
  //     cyphered = data;
  //   });
  //   console.log(cyphered);
  //   expect(processor(3)).toBe(cyphered);
  // });
  test("should ", () => {
    expect(processor(-1, "hello")).toBe("gdkkn");
  });
  test("should ", () => {
    expect(processor(-1, "aello")).toBe("zdkkn");
  });
});

File: go/index_test.go
package main

import (
	"fmt"
	"testing"
)

func TestCypher(t *testing.T) {

	tables := []struct {
		x []byte
		y int
		n string
	}{
		{[]byte("hello"), 2, "jgnnq"},
		{[]byte("helloy"), 3, "khoorb"},
		{[]byte("hello"), -1, "gdkkn"},
		{[]byte("aello"), -1, "zdkkn"},
		{[]byte("hel.l,o"), 2, "jgn.n,q"},
	}

	for _, table := range tables {
		cyphered := cypher(table.x, table.y)
		if cyphered.String() != table.n {
			fmt.Printf("error")
			t.Errorf("Expected '%s' but got '%s'", table.n, cyphered)
		}
	}

}
